lib/async/redis/sentinels.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+module Async
+	module Redis
+		class SentinelsClient < Client
+			def initialize(master_name, sentinels, role = :master, protocol = Protocol::RESP2, **options)
+				@master_name = master_name
+				@sentinel_endpoints = sentinels.map do |sentinel|
+					Async::IO::Endpoint.tcp(sentinel[:host], sentinel[:port])
+				end
+				@role = role
+
+				@protocol = protocol
+				@pool = connect(**options)
+			end
+
+			private
+
+			# Override the parent method. The only difference is that this one needs
+			# to resolve the master/slave address.
+			def connect(**options)
+				Async::Pool::Controller.wrap(**options) do
+					address = resolve_address
```

Comment:
Rather than using a hash, which I know probably feels more explicit, but it's also more expensive. I'd suggest either:

- Return an endpoint instance.
- Use an array of `[host, port]` as in `host, port = resolve_address`.

Reply:
I've changed this to return endpoints.
I think the second option hurts readability a bit because while checking the code you need to remember what each index represents.